DcNotificationService/NotificationService.swift

```diff
@@ -0,0 +1,98 @@
+import UserNotifications
+import DcCore
+
+class NotificationService: UNNotificationServiceExtension {
+    let dcAccounts = DcAccounts.shared
+
+    override func didReceive(_ request: UNNotificationRequest, withContentHandler contentHandler: @escaping (UNNotificationContent) -> Void) {
+        guard let bestAttemptContent = (request.content.mutableCopy() as? UNMutableNotificationContent) else { return }
+
+        if UserDefaults.mainAppRunning {
+            contentHandler(silenceNotification(bestAttemptContent))
+            return
+        }
+        UserDefaults.setNseFetching()
+
+        // as we're mixing in notifications from accounts without PUSH and we cannot add multiple notifications,
+        // it is best to move everything to the same thread - and set just no threadIdentifier
+
+        dcAccounts.openDatabase(writeable: false)
+        let eventEmitter = dcAccounts.getEventEmitter()
+
+        if !dcAccounts.backgroundFetch(timeout: 25) {
+            UserDefaults.setNseFetching(false)
+            contentHandler(bestAttemptContent)
+            return
+        }
+        UserDefaults.setNseFetching(false)
+
+        var messageCount = 0
+        var uniqueChats: [String: String] = [:]
+        while true {
+            guard let event = eventEmitter.getNextEvent() else { break }
+            if event.id == DC_EVENT_ACCOUNTS_BACKGROUND_FETCH_DONE { break }
+            if event.id == DC_EVENT_INCOMING_MSG {
+                let dcContext = dcAccounts.get(id: event.accountId)
+                let chat = dcContext.getChat(chatId: event.data1Int)
+                if !UserDefaults.standard.bool(forKey: "notifications_disabled") && !chat.isMuted {
+                    let msg = dcContext.getMessage(id: event.data2Int)
+                    let sender = msg.getSenderName(dcContext.getContact(id: msg.fromContactId))
+                    bestAttemptContent.title = chat.isGroup ? chat.name : sender
+                    bestAttemptContent.body = (chat.isGroup ? "\(sender): " : "") + (msg.summary(chars: 80) ?? "")
+                    bestAttemptContent.userInfo["account_id"] = dcContext.id
```

Comment:
We could think about moving these strings into an enum of constants as we use these in several places

Reply:
yeah, maybe, i would not  be against it. but i also think it is also good as is. from all issues, we rarely have problems with this kind of stuff :)